From f6f0d88b5e90ea9af2f59af4b9a39356da405335 Mon Sep 17 00:00:00 2001
From: dnagasun <devamekalai.nagasundaram@intel.com>
Date: Fri, 1 Nov 2019 11:00:48 +0800
Subject: [PATCH] [VP] Fix GPU hang

Fix GPU hang on EHL, disable AVS since no AVS sampler

[Internal]
Platforms: Gen11
OS: Linux
Feature impact: CSC_Scal_Rot_Mirr
Resolves: N/A
Related-to: N/A
Klocwork: PASS
IP Scan: PASS

Author	Chen, Jason K <jason.k.chen@intel.com>
Signed-off-by: dnagasun <devamekalai.nagasundaram@intel.com>
---
 .../agnostic/common/vp/hal/vphal_render_composite.cpp       | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp b/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp
index 2dca478a..a5bf51f5 100644
--- a/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp
+++ b/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp
@@ -2477,6 +2477,12 @@ void CompositeState::SetScalingMode(
         pSource->ScalingMode = VPHAL_SCALING_BILINEAR;
     }
 
+    // Fix GPU Hang on EHL, since EHL has no AVS sampler
+    if (MEDIA_IS_SKU(m_pSkuTable, FtrDisableVEBoxFeatures))
+    {
+        pSource->ScalingMode = VPHAL_SCALING_BILINEAR;
+    }
+
     // WA for multilayer P010 AVS+3D one single pass corruption hw issue
     if (uSourceCount > 1 &&
         pSource->Format == Format_P010)
-- 
2.23.0

